tests/cases/conformance/types/intersection/intersectionTypeInference2.ts

```diff
@@ -0,0 +1,15 @@
+declare function f<T>(x: { prop: T }): T;
+
+declare const a: { prop: string } & { prop: number };
+declare const b: { prop: string & number };
+
+f(a);  // string & number

```

Comment:
Consider using more realistic types for this test given that we've already been thinking about removing uninhabitable types.